### ISSUE TITLE
Unable to set-UDElement - can't seem to determine the scope where you can

### ISSUE THREAD

**Issue by Phyziik** [ironmansoftware/universal-dashboard#49]
Hi,
I'm trying to modify the elements of my custom table but I can't seem to interact with element with their ID : 
```
New-UDCollapsibleItem -Title "Testing cycle" -Active -Endpoint {
         New-UDElement -Tag "table" -Attributes @{border="1"} -Endpoint {
            New-UDElement -tag "thead" -Content {
               New-UDElement -Tag "tr" -Content {
                  New-UDElement -Tag "td" -id "currentCell" -Content {"Initial value"}
               }
            }
         }
      
      
         New-UDInput -Title "Press me" -SubmitText "Granting your wish" -DebugEndpoint -Endpoint{
            param($value)
            #start notepad
            Set-UDElement -Id "currentCell" -content {"Changed this"}
         }
      }
```
Here is a simplified version of what I want to do. I tried debuging my runspace, and once inside, I tried doing some Get-UDElement using my ID, and I don't get anything.

Thanks

**Comment by Phyziik** [comment]
This is the type of errors I get. I'm trying to do `Set-UDElement` in different location in my code, and none of those seem to have access to the scope of my custom table.

> Microsoft.AspNetCore.SignalR.HubEndPoint Failed to invoke hub method

**Comment by Phyziik** [comment]
And I don't know if this is related too, but when you use `-Endpoint` on `New-UDElement`, I get a constant `Loading` message. I tried adding and removing `-autoRefresh -RefreshInterval 5` and it stays the same, never showing the actual message inside my endpoint.

**Comment by adamdriscoll** [comment]
@Phyziik Could you please upload a full log?

**Comment by Phyziik** [comment]
Here are my logs. I took the liberty of my making a debug dashboard, just to remove all the issues that could be related to my code.

[DebugDashboard.txt](https://github.com/ironmansoftware/universal-dashboard/files/1847472/DebugDashboard.txt)
[logs_debug.txt](https://github.com/ironmansoftware/universal-dashboard/files/1847473/logs_debug.txt)

**Comment by Phyziik** [comment]
I'd say that my biggest issue with coding this table is figuring out the way you programmed the Content vs Endpoint. You mentionned that Content = Static and Endpoint = Dynamic, so I figured that if I wanted to refresh my table automatically, I'd use the Endpoint. So I referred to your example ([UD-Clock](https://github.com/ironmansoftware/ud-clock/blob/master/dashboard.ps1)). I played around with the -endpoint on new-UDelement, which caused `Loading` to show at all time. Then, I tried a simple test where every 5 seconds, a value should increment, and modify the content of the table dynamically, like the following : 

```
New-UDCollapsible -BackgroundColor '#252525' -Items {
      New-UDCollapsibleItem -Title "Acceptance" -BackgroundColor '#FF1c1c1c' -FontColor "#FFFFFF" -Active -Endpoint {
         New-UDColumn -AutoRefresh -RefreshInterval 5 -Endpoint {
            $value = Get-TestValue
            Set-UDElement -Id "Test1_status" -Content {"New value :$value"}
 
         }
         New-UDRow -Columns {
            New-UDColumn -Size 12 -Content {
               New-UDCard @Colors -Content {
                  New-UDElement -Tag "table" -Attributes @{border="1"} -Content {
                     New-UDElement -tag "thead" -Content {
                        New-UDElement -tag "tr" -Content {
                           New-UDElement -tag "td" -Id "NameHeader" -Attributes @{class="table-cell"} -Content {"Name"}
                           New-UDElement -tag "td" -Attributes @{class="table-cell"} -Content {"Description"}
                           New-UDElement -tag "td" -Attributes @{class="table-cell"} -Content {"Status"}
                           New-UDElement -tag "td" -Attributes @{class="table-cell"} -Content {"Comments"}
                           New-UDElement -tag "td" -Attributes @{class="table-cell"} -Content {" "}
                        }
                     }
                     New-UDElement -tag "tbody" -Content {
                        New-UDElement -Tag "tr" -Content {
                           New-UDElement -Tag "td" -Content {"Test1"}
                           New-UDElement -Tag "td" -Content {"Doing test 1"}
                           New-UDElement -Tag "td" -Id "Test1_status" -Content {"Test1 Status"}
                           New-UDElement -Tag "td" -Id "Test1_comments" -Content {"Test1 comments"}
                           New-UDElement -Tag "td" -Attributes @{style = @{'text-align'='center'}} -Content {
                              New-UDElement -Id "Test1_button" -Tag "a" -Attributes @{class='btn'} -Content {"Run"}
                           }
                        }

                     }
                  }
               }
            }
         } 
      }
   }
```
and here is my endpointinitialization for my dashboard :
```
... -EndpointInitializationScript {
   $script:incrementalValue = 0
   
   #for testing purpose
   function Get-TestValue 
   {
      $script:incrementalValue++
      return $script:incrementalValue
   }
}
```
Here are the logs : 
[logs_debug.txt](https://github.com/ironmansoftware/universal-dashboard/files/1847539/logs_debug.txt)

**Comment by adamdriscoll** [comment]
I can reproduce this. This seems like it's a bug. I'll let you know what I find.

**Comment by Phyziik** [comment]
Great, can't wait to see the result. Related to the New and Set-UDElement, there is the issue of the endpoint and the loading message?
And is there any way to debug whatever is happening inside of a onClick of a button? We have DebugEndpoint, but we can't use this on a New-UDElement. I have a feeling that New-UDButton possibly will have the actions under an endpoint param, but I rather ask to know if there is any possible solution with UDElement.

Thanks

**Comment by Phyziik** [comment]
After more testing, and still the same bug affecting, I found out that the Set-UDElement only works when I put it oustide of my New-UDCollapsible. I'm still trying to understand where it works and where it doesn't. All I know is that it doesn't work in all scopes.

**Comment by adamdriscoll** [comment]
It should work in all scopes so that is a bug with New-UDCollapsible. I will be replacing the implementation of that with the 1.6.0 release so we should see this issue fixed with that. 

In terms of debugging event handlers (like onClick), it should work but it isn't implemented at the moment. I will add a separate issue for that.

**Comment by adamdriscoll** [comment]
This has been fixed in the upcoming 1.6.0 release.
